docs/design.md

```diff
@@ -1,5 +1,33 @@
 # Design
 
+From repository `task-git` to `tektoncd-catalog` `p` branch.
+
+- In `task-git` repository
+	- I have a "configuration file" (`catalog.yaml` ?) that lists Tasks and Pipelines from the repository that I want to release
+		- This file can be generated, updated, … 
+	- When I do the release, I want to issue one command (`catalog-cd release`)
+		- it mutates the resources to add the version annotation
+		- it generates the final `catalog.yaml` with hash, digest, signature, …
+		- it packages the tasks and pipelines in a `tekton-resources.tar.gz` tarball (with READMEs for documentation)
+		- it (optionally) create, push the tag, create a GitHub release and attach content to it
```

Comment:
Here I expect users to rely on `gh` directly, so `catalog-cd release` will do all the heavy lifting for the user be able to only upload the release.

Reply:
In my head, it was more, something similar to `goreleaser`. Quite some project using `goreleaser` do not create any GitHub release using the UI or `gh`, just rely on the `gorelease` tool. The idea was kind-of the same 🙃